planner/core/plan_cost.go

```diff
@@ -250,32 +250,114 @@ func (p *PhysicalIndexScan) GetPlanCost(taskType property.TaskType) (float64, er
 
 // GetPlanCost calculates the cost of the plan if it has not been calculated yet and returns the cost.
 func (p *PhysicalIndexJoin) GetPlanCost(taskType property.TaskType) (float64, error) {
```

Comment:
It seems all of the `GetPlanCost` functions for the different physical joins are very similar. Can we implement the `GetPlanCost` for `basePhysicalJoin` only, so we don't need to implement it for the different physical joins?

Reply:
I thought about this, but `HashJoin` and `MergeJoin` also inherit it, so I'm afraid it could be confused if we implement

Reply:
I thought about this, but I'm afraid it could be confused since `HashJoin` and `MergeJoin` also inherit it, so I choose to implement them separately in the end.